modules/rest-api/src/main/java/org/apache/ignite/internal/rest/api/transaction/TransactionApi.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.rest.api.transaction;
+
+import static io.swagger.v3.oas.annotations.media.Schema.RequiredMode.REQUIRED;
+import static org.apache.ignite.internal.rest.constants.MediaType.APPLICATION_JSON;
+
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Delete;
+import io.micronaut.http.annotation.Get;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.ArraySchema;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.Collection;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import org.apache.ignite.internal.rest.api.Problem;
+
+/**
+ * API for managing transaction.
+ */
+@Controller("/management/v1/transactions/")
+@Tag(name = "transactions")
+public interface TransactionApi {
+    /**
+     * Retrieves all in progress transactions.
+     *
+     *
+     * @return A collection of all in progress transactions.
+     */
+    @Operation(summary = "Retrieve all in progress transactions", description = "Fetches all in progress transactions.")
+    @ApiResponse(
+            responseCode = "200",
+            description = "Successfully retrieved in progress transactions.",
+            content = @Content(mediaType = APPLICATION_JSON, array = @ArraySchema(schema = @Schema(implementation = Transaction.class)))
+    )
+    @Get
+    CompletableFuture<Collection<Transaction>> transactions();
+
+    /**
+     * Retrieves the state of a specific transaction.
```

Comment:
It seems the "state" here is not the best word because transaction already has a field "state" but we return the "whole" transation.